src/main/java/com/thebois/models/beings/AbstractBeing.java

```diff
@@ -153,8 +155,48 @@ protected void setPath(final Collection<Position> path) {
         return Optional.of(path.peek().deepClone());
     }
 
+    protected Optional<Position> getFinalDestination() {
+        if (path.isEmpty()) return Optional.empty();
+        return Optional.of(path.firstElement().deepClone());
+    }
+
     protected IPathFinder getPathFinder() {
         return pathFinder;
     }
 
+    protected IFinder getFinder() {
+        return this.finder;
+    }
+
+    protected void setFinder(final IFinder finder) {
+        this.finder = finder;
+    }
+
+    protected Position nearestPositionNextTo(final Position destination) {
+        final int[][] positionsToCheck = {
+            {-1, -1}, {0, -1}, {1, -1},
+            {-1, 0}, {1, 0},
+            {-1, 1}, {0, 1}, {1, 1},
+            };
+        boolean firstLoop = true;
+
+        Position nearestPosition = new Position();
+        Position lastPosition;
+
+        for (int i = 0; i < positionsToCheck.length; i++) {
+            final float xPos = destination.getPosX() + positionsToCheck[i][0];
+            final float yPos = destination.getPosY() + positionsToCheck[i][1];
+            lastPosition = new Position(xPos, yPos);
+            if (firstLoop) {
+                nearestPosition = lastPosition;
+                firstLoop = false;
+            }
+
+            if (getPosition().manhattanDistanceTo(nearestPosition)
+                > getPosition().distanceTo(lastPosition)) {
```

Comment:
question

How come you're comparing a manhattan distance to a euclidian distance? I don't see the reason for not using manhattan distance for both. And if there's a good reason, you should probably add a comment explaining that reason.